syft/source/file_type.go

```diff
@@ -34,3 +37,15 @@ func newFileTypeFromTarHeaderTypeFlag(flag byte) FileType {
 	}
 	return UnknownFileType
 }
+
+// TODO: fill in more types from mod...
+func newFileTypeFromMode(mod os.FileMode) FileType {
+	switch {
+	case mod&os.ModeSymlink == os.ModeSymlink:
```

Comment:
Ooo I haven't seen the ampersand used this way before, what's going on here?

Reply:
here we're taking the os.FileMode, which is a `uint32`, and performing a bitwise "and" on the given mode against a another `uint32` that has a single bit set in order to check if a specific bit is set (in this case to see if the symlink bit is set). This could also be written as `case mod&os.ModeSymlink != 0`.